contracts/commands/BaseDMACommand.sol

```diff
@@ -0,0 +1,140 @@
+    error CallerNotAutomationBot(address caller);
+    error InvalidOperationHash(bytes32 operationHash);
+    error InvalidSelector(bytes4 selector);
+
+    using RatioUtils for uint256;
+
+    address public immutable bot;
+    address public immutable self;
+    address public immutable weth;
+    IOperationExecutor public immutable operationExecutor;
+
+    string private constant AUTOMATION_BOT = "AUTOMATION_BOT_V2";
+    string private constant OPERATION_EXECUTOR = "OperationExecutor_2";
+    string private constant WETH = "WETH";
+
+    uint256 public constant MIN_ALLOWED_DEVIATION = 50; // corrresponds to 0.5%
+
+    /**
+     * @dev Constructor function.
+     * @param _serviceRegistry The address of the service registry contract.
+     */
+    constructor(IServiceRegistry _serviceRegistry) {
+        // Validate service registry address
+        if (address(_serviceRegistry) == address(0)) {
+            revert EmptyAddress("service registry");
+        }
+        // Get the address of the automation bot from the service registry
+        bot = _serviceRegistry.getRegisteredService(AUTOMATION_BOT);
+        // Validate automation bot address
+        if (bot == address(0)) {
+            revert EmptyAddress("bot");
+        }
+
+        // Get the address of the operation executor from the service registry
+        operationExecutor = IOperationExecutor(
+            _serviceRegistry.getRegisteredService(OPERATION_EXECUTOR)
+        );
+        // Validate operation executor address
+        if (address(operationExecutor) == address(0)) {
+            revert EmptyAddress("operation executor");
+        }
+
+        // Get the address of the WETH token from the service registry
+        weth = _serviceRegistry.getRegisteredService(WETH);
+        // Validate WETH address
+        if (weth == address(0)) {
+            revert EmptyAddress("weth");
+        }
+    }
+
+    /**
+     * @dev Checks if the provided deviation is valid.
+     * @param deviation The deviation value to check.
+     * @return A boolean indicating whether the deviation is valid or not.
+     */
+    function deviationIsValid(uint256 deviation) internal pure returns (bool) {
+        return deviation >= MIN_ALLOWED_DEVIATION;
+    }
+
+    /**
+     * @dev Checks if the provided base fee is valid.
+     * @param maxAcceptableBaseFeeInGwei The maximum acceptable base fee in Gwei.
+     * @return A boolean indicating whether the base fee is valid or not.
+     */
+    function baseFeeIsValid(uint256 maxAcceptableBaseFeeInGwei) internal view returns (bool) {
+        return block.basefee <= maxAcceptableBaseFeeInGwei * (10 ** 9);
+    }
+
+    /**
+     * @dev Validates the trigger type.
+     * @param triggerType The actual trigger type.
+     * @param expectedTriggerType The expected trigger type.
+     */
+    function validateTriggerType(uint16 triggerType, uint16 expectedTriggerType) internal pure {
+        if (triggerType != expectedTriggerType) {
```

Comment:
Minor: I see we are checking if the trigger type is valid in `AaveV3BasicSellCommandV2.isTriggerDataValid`. I'd suggest to add to `BaseDMACommand` a function `_isTriggerTypeValid(...) internal` and then rewrite this function as:
```suggestion
        if (!_isTriggerTypeValid(triggerType, expectedTriggerType)) {
```
so we always use the same logic when comparing triggers, and if the comparison must be updated, it can be done in a single place